Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1201,49 +1201,55 @@ namespace Babylon
 
                 // The constructor is either sent a BABYLON.Vector3, {}, an XRRigidTransform, or {x,y,z,w},{x,y,z,w}
                 if (argLength > 0 && info[0].IsObject())
                 {
-                    auto argumentObject = info[0].As<Napi::Object>();
+                    auto argumentObject{info[0].As<Napi::Object>()};
 
-                    XRRigidTransform* transform = XRRigidTransform::Unwrap(argumentObject);
+                    XRRigidTransform* transform{XRRigidTransform::Unwrap(argumentObject)};
                     if (transform != nullptr)
                     {
-                        // The value passed in to the contructor is an XRRigidTransform
-                        xr::Pose pose = transform->GetNativePose();
+                        // The value passed in to the constructor is an XRRigidTransform
+                        xr::Pose pose{transform->GetNativePose()};
                         tempVals.Origin = pose.Position;
 
                         // Grab forward direction from quaternion
                         tempVals.Direction.X = 2 * ((pose.Orientation.X * pose.Orientation.Z) + (pose.Orientation.W * pose.Orientation.Y));
                         tempVals.Direction.Y = 2 * ((pose.Orientation.Y * pose.Orientation.Z) - (pose.Orientation.W * pose.Orientation.X));
                         tempVals.Direction.Z = 1 - (2 * ((pose.Orientation.X * pose.Orientation.X) + (pose.Orientation.Y * pose.Orientation.Y)));
                     }
-
-                    if (argumentObject.Has("x"))
-                    {
-                        tempVals.Origin.X = argumentObject.Get("x").ToNumber().FloatValue();
-                    }
-                    if (argumentObject.Has("y"))
-                    {
-                        tempVals.Origin.Y = argumentObject.Get("y").ToNumber().FloatValue();
-                    }
-                    if (argumentObject.Has("z"))
+                    else
                     {
-                        tempVals.Origin.Z = argumentObject.Get("z").ToNumber().FloatValue();
+                        if (argumentObject.Has("x"))
+                        {
+                            tempVals.Origin.X = argumentObject.Get("x").ToNumber().FloatValue();
+                        }
+                        if (argumentObject.Has("y"))
+                        {
+                            tempVals.Origin.Y = argumentObject.Get("y").ToNumber().FloatValue();
+                        }
+                        if (argumentObject.Has("z"))
+                        {
+                            tempVals.Origin.Z = argumentObject.Get("z").ToNumber().FloatValue();
+                        }
+                        if (argumentObject.Has("w") && argumentObject.Get("w").ToNumber().FloatValue() != 1.0)
+                        {
+                            Napi::Error::New(info.Env(), "TypeError: w-axis provided for XRRay's Origin is not 1").ThrowAsJavaScriptException();
```

Comment:
nit: alternative way
```suggestion
                            throw Napi::Error::New(info.Env(), "TypeError: w-axis provided for XRRay's Origin is not 1");
```

Reply:
Sounds good, I'll use that instead, since it's cleaner.